api/backend/infrastructure/osm/street_name_parser.py

```
@@ -0,0 +1,62 @@
+
+def parse_name(nodes_list):
+    """Infer a street name for the provided OSM way nodes using DaData geolocation."""
+    dadata = Dadata(_get_dadata_token())
+
+    length = len(nodes_list)
+    for i in range(0, length - 1):
+        try:
+            lat = (nodes_list[i].location.lat + nodes_list[i + 1].location.lat) / 2
+            lon = (nodes_list[i].location.lon + nodes_list[i + 1].location.lon) / 2
+        except Exception:
+            continue
```

Ruff (0.14.5)

`try`-`except`-`continue` detected, consider logging the exception

(S112)